pkg/testhelpers/clock.go

```diff
@@ -0,0 +1,16 @@
+package testhelpers
+
+import (
+	"time"
+
+	"github.com/facebookgo/clock"
+)
+
+// NewMockClock returns a mock clock that returns the given time for Now()
+func NewMockClock(now time.Time) *clock.Mock {
```

Comment:
Rather than creating a new mock clock every time, can we just update the existing clock?

`
type SettableClock struct {
MockClock
}

func (*c SettableClock) Set(now time.Time) {
for !c.MockClock.Now().Equal(now) {
		c.MockClock.Add(now.Sub(newClock.Now()))
	}
`

Reply:
Constructor is just:
```
func NewSettableClock() *SettableClock {
    return SettableClock{clock.NewMock()}
}
```

Reply:
Actually, with composition, you don't even need to do `c.MockClock.Now()`, it's just `c.Now()` and `c.Add()`... A few other syntax typos in there too.